dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/DataSourceServiceImpl.java

```diff
@@ -443,4 +447,62 @@ private String getHiddenPassword() {
         return result;
     }
 
+    /**
+     * query datasource list created by user
+     *
+     * @param userId user id
+     * @return datasource list
+     */
+    @Override
+    public List<DataSource> queryCreatedByUser(int userId) {
+        return dataSourceMapper.selectList(Wrappers.<DataSource>lambdaQuery().eq(DataSource::getUserId, userId));
+    }
+
+    /**
+     * transfer datasource list owned by the user
+     *
+     * @param transferredUserId transferred user id
+     * @param receivedUserId received user id
+     * @param transferredIds transferred ids
+     * @return transfer result code
+     */
+    @Override
+    public Map<String, Object> transferOwnedData(int transferredUserId, int receivedUserId, List<Integer> transferredIds) {
+        Map<String, Object> result = new HashMap<>();
+
+        List<DataSource> dataSources = dataSourceMapper.selectList(Wrappers.<DataSource>lambdaQuery()
+                .eq(DataSource::getUserId, transferredUserId)
+                .in(DataSource::getId, transferredIds)
+        );
+        Set<Integer> realDataSourceIds = dataSources.stream().map(DataSource::getId).collect(Collectors.toSet());
+        // update datasource owner
+        int updatedDatasourceNum = dataSourceMapper.update(null, Wrappers.<DataSource>lambdaUpdate()
```

Comment:
As if this updating action could be combined with 'dataSourceMapper.selectList'. Seems like it below:

int updatedDatasourceNum = dataSourceMapper.update(null, Wrappers.<DataSource>lambdaUpdate()
                .set(DataSource::getUserId, receivedUserId)
                .set(DataSource::getUpdateTime, new Date())
                .in(DataSource::getId, realDataSourceIds)
                .eq(DataSource::getUserId, transferredUserId);

Reply:
Yes, it has been modified.